### DISCUSSION TITLE
How to programmatically pin rich text toolbar

### DISCUSSION THREAD

**Discussion by ilia4bit** [wagtail/wagtail#11409] [Q&A] [answered]
If you want to pin a rich-text toolbar first you need to write something in the field, then press `ctrl + A` to see a rich-text toolbar with full features (because Wagtail doesn't load all features in a stream block-like interface) and press the `pin` icon.
<img width="599" alt="Screenshot 1402-10-12 at 6 15 59 in the evening" src="https://github.com/wagtail/wagtail/assets/69800165/498a829a-c272-4919-ab43-a576142341a0">
<img width="587" alt="Screenshot 1402-10-12 at 6 16 47 in the evening" src="https://github.com/wagtail/wagtail/assets/69800165/9ef6c5f1-ab12-4663-ad97-7139bcd2f94a">

**Is there any way to pin the rich-text toolbar by code?**
can I write something like this:
```python
class MyPage(Page):
 content = RichTextField()
 
 content_panels = [
  FieldPanel("content", classname="pin")
]
```

**Comment by thibaudcolas** [comment, marked as answer]
@ilia4bit good question – there is no way currently, though with the right amount of CSS overrides nothing is impossible. This isn’t officially supported but you could do this if you’re ok with changing the default for all fields:

```js
// If the toolbar’s theme isn’t set, default to sticky.
if (!window.localStorage.getItem('wagtail:draftail-toolbar')) {
  window.localStorage.setItem('wagtail:draftail-toolbar', 'sticky');
}
```

**Reply by ilia4bit** [reply]
Thank you for your response. I use the `insert_editor_js` hook and it works. but without if statement. 
Correct code:
- `wagtail_hooks.py`
```python
from django.utils.html import format_html
from django.templatetags.static import static
from wagtail import hooks


@hooks.register("insert_editor_js")
def editor_js():
    return format_html('<script src="{0}"></script>', static("static/js/pin-toolbar.js"))
```
- `pin-toolbar.js`
```js
window.localStorage.setItem("wagtail:draftail-toolbar", "sticky");
```

**Reply by thibaudcolas** [reply]
I wouldn’t recommend doing this without the `if` statement as it means it’ll now be impossible for CMS users to ever change this setting.

**Reply by ilia4bit** [reply]
I tried but it doesn't work at least for me.
